client/rest/src/main/java/org/opensearch/client/RestClient.java

```diff
@@ -343,6 +343,7 @@ private ResponseOrResponseException convertResponse(InternalRequest request, Nod
         if (isSuccessfulResponse(statusCode) || request.ignoreErrorCodes.contains(response.getStatusLine().getStatusCode())) {
             onResponse(node);
             if (request.warningsHandler.warningsShouldFailRequest(response.getWarnings())) {
+                logger.warn(Arrays.toString(response.getWarnings().toArray()));
```

Comment:
log-and-throw is generally an anti-pattern. We really can't know whether logging is the right thing to do in all contexts for a general purpose client like this, so it is usually better to let whatever handles the exception make the logging decision. In the context of the tests it is definitely a good idea to log these details. Is it possible to handle this exception in the test runner somewhere and log it there?

Reply:
I agree with @andrross here, wdyt @dreamer-89?

Reply:
Thanks @andrross @dblock for the feedback. Yes, I also agree with @andrross here. 

The test `recoverReplica` directly calls the rest client via parent class `OpenSearchRestTestCase` method. So, there are couple of options for adding log statement;
1.  Inside parent class `OpenSearchRestTestCase` 
2. Individual test cases.

#1 seems better option as its minor change and covers all cases.

I think since issue the original issue is fixed in https://github.com/opensearch-project/OpenSearch/issues/2334; we can hold onto this change for now ?